Feature/typescript

graysonhicks opened the pull request:
DRAFT PR to fix #207 

Draft PR reliant on #238 first.  

Adds TS and types with corresponding clean-up.

graysonhicks committed:
remove type references

rewrite in typescript

start

fixed all types but one

tsc build command and fix map

deprecated apis and peerdeps

graysonhicks commented:
I tested this on my site and it works as expected @oorestisime

oorestisime commented on README.md:
````diff
@@ -56,29 +52,6 @@ plugins: [
 ]
 ```
-
-### Public scraping for a user's profile
````
can we keep those deprecated for reference please? i dont have any other docs page :D

oorestisime commented on package.json:
```diff
@@ -23,7 +27,11 @@
     "eslint-plugin-prettier": "3.1.4",
     "eslint-plugin-react": "7.20.6",
-    "prettier": "2.1.2"
+    "gatsby": "^4.24.7",
```
what about gatsby 5 compat?

graysonhicks replied:
Yea we probably just need it as a peer dependency.

oorestisime reviewed (commented):
can you rebase please? Also left some comments

graysonhicks committed:
merge conflicts

graysonhicks commented:
Okay, sorry for the delay.  Rebased this and added the deprecation notes back.  Tested again on my site and works as expected.  I noticed in the example dir, there may be an issue with `hashtags` but not sure what the expectation is there.
